### PR TITLE
Add runtime value interface to m3cluster

### PR DESCRIPTION
**Pull request by xichen2020**
cc @cw9 @jeromefroe @martin-mao 

This PR adds the runtime value interface to m3cluster, which provides the ability to watch / unwatch updates and process the updates using the configurable unmarshalling function and processing function.

### CODE REVIEW

File: kv/util/runtime/value_test.go [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+
+	_, rv := testValueWithMockStore(ctrl)
+	rv.version = 3
+	rv.unmarshalFn = func(v kv.Value) (interface{}, error) { return v, nil }
+
+	require.NoError(t, rv.updateWithLock(mockValue{version: 2}))
+	require.Equal(t, 3, rv.version)
+}
+
+type mockValue struct {
```
Comment (cw9) [inline kv/util/runtime/value_test.go]: can this be replaced with mem.NewValue() ?

Comment (xichen2020) [inline kv/util/runtime/value_test.go]: Sure thing

Comment (xichen2020) [inline kv/util/runtime/value_test.go]: Sure thing.

File: kv/util/runtime/value.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+	"sync"
+	"time"
+
+	"github.com/m3db/m3cluster/kv"
+	"github.com/m3db/m3x/log"
+)
+
+const (
+	// DefaultVersion is the default version of an uninitialized kv value.
+	DefaultVersion = 0
```
Comment (xichen2020) [inline kv/util/runtime/value.go]: Talked offline, will move to `kv/`

File: kv/util/runtime/options.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	unmarshalFn      UnmarshalFn
+	processFn        ProcessFn
+}
+
+// NewOptions creates a new set of options.
+func NewOptions() Options {
+	return &options{
+		instrumentOpts:   instrument.NewOptions(),
+		initWatchTimeout: defaultInitWatchTimeout,
+		kvStore:          mem.NewStore(),
```
Comment (xichen2020) [inline kv/util/runtime/options.go]: Hm using the memstore feels like a reasonable default to me, leaving it nil is slightly better but we are back to the same place if the users don't call `Validate`.

**Pull request merged**